src/gui/SetupDialog.cpp

```diff
@@ -1319,7 +1340,176 @@ void SetupDialog::setBackgroundPicFile(const QString & backgroundPicFile)
 }
 
 
+void SetupDialog::setLabels1Dir(const QString & labels1Dir)
+{
+	m_labels1Dir = labels1Dir;
+}
+
+void SetupDialog::openLabels1Dir()
+{
+	QString new_file = FileDialog::getOpenFileName(this,
+		tr("Choose your labels file"), m_labels1Dir, "Text files (*.txt)");
+
+	if (!new_file.isEmpty())
+	{
+		bool fileCorrect = 1;
+		QFile labelsFile(new_file);
+		labelsFile.open(QIODevice::ReadOnly);
+		if (!labelsFile.isOpen()) return ;
+		
+		QTextStream stream(&labelsFile);
+		QString line = stream.readLine();
+		QRegExp num("\\d*");
+		while (!line.isNull()) {
+			if(line == ""){
+				break;
+			}
+			if(line.mid(0, 2) == "//"){
+				//Comment line
+			} else if (num.exactMatch(line.mid(0,1))){
+				if (num.exactMatch(line.mid(1,1))){
+					if (num.exactMatch(line.mid(2,1))){
+						int number = line.mid(0,3).toInt();
+						if (number > 127){
+							fileCorrect = 0;			
+							break;
+						} else if (line.mid(3,1) != " " || line.mid(3,2) == "  "){
+							fileCorrect = 0;
+							break;
+						}
+					} else if (line.mid(2,1) != " " || line.mid(2,2) == "  "){
+						fileCorrect = 0;
+						break;
+					}		
+				} else if (line.mid(1,1) != " " || line.mid(1,2) == "  ") {
+					fileCorrect = 0;
+					break;
+				}	
+			} else {
+				fileCorrect = 0;
+				break;
+			}
+			line = stream.readLine();
+		}	
+		if(fileCorrect){
+			m_labels1DirLineEdit->setText(new_file);
+		}	
+	}
+}
+
+void SetupDialog::setLabels2Dir(const QString & labels2Dir)
+{
+	m_labels2Dir = labels2Dir;
+}
+
+void SetupDialog::openLabels2Dir()
+{
+	QString new_file = FileDialog::getOpenFileName(this,
+		tr("Choose your labels file"), m_labels2Dir, "Text files (*.txt)");
+
+	if (!new_file.isEmpty())
+	{
+		bool fileCorrect = 1;
+		QFile labelsFile(new_file);
+		labelsFile.open(QIODevice::ReadOnly);
+		if (!labelsFile.isOpen()) return ;
+		
+		QTextStream stream(&labelsFile);
+		QString line = stream.readLine();
+		QRegExp num("\\d*");
+		while (!line.isNull()) {
```

Comment:
These loops and ifs within these blocks should follow the [Coding Conventions](https://github.com/LMMS/lmms/wiki/Coding-Conventions).

Reply:
Reviewed all changes, and made them follow Coding Conventions.